File: llm-context-debugger/test_dashboard.py
"""Playwright test script for dashboard screenshots and verification."""

import os
from pathlib import Path
from playwright.sync_api import sync_playwright

SCREENSHOTS_DIR = Path(__file__).parent / "screenshots"
SCREENSHOTS_DIR.mkdir(exist_ok=True)
BASE_URL = "http://localhost:8088"


def main():
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page(viewport={"width": 1400, "height": 900})

        # 1. Empty state
        page.goto(BASE_URL)
        page.wait_for_timeout(1000)
        page.screenshot(path=str(SCREENSHOTS_DIR / "01_empty_state.png"))
        print("[OK] Screenshot: empty state")

        # 2. Load demo data
        page.click("text=Load Demo")
        page.wait_for_timeout(2000)
        page.screenshot(path=str(SCREENSHOTS_DIR / "02_demo_overview.png"))
        print("[OK] Screenshot: demo overview")

        # 3. Click on call #4 (the one with warning)
        items = page.query_selector_all(".call-item")
        if len(items) >= 3:
            items[-1].click()  # Last item should have the warning
            page.wait_for_timeout(1000)
            page.screenshot(path=str(SCREENSHOTS_DIR / "03_warning_call.png"))
            print("[OK] Screenshot: warning call")

        # 4. Messages tab
        page.click("text=Messages")
        page.wait_for_timeout(500)
        page.screenshot(path=str(SCREENSHOTS_DIR / "04_messages_tab.png"))
        print("[OK] Screenshot: messages tab")

        # 5. Diff tab - select call #2 overview first
        items = page.query_selector_all(".call-item")
        if len(items) >= 2:
            items[1].click()  # Second call
            page.wait_for_timeout(1000)

        page.click("text=Diff")
        page.wait_for_timeout(500)
        page.click("text=Compare")
        page.wait_for_timeout(1000)
        page.screenshot(path=str(SCREENSHOTS_DIR / "05_diff_view.png"))
        print("[OK] Screenshot: diff view")

        # 6. Verify key elements exist
        page.click("text=Overview")
        page.wait_for_timeout(500)
        checks = {
            "bar_chart": page.query_selector(".token-chart") is not None,
            "treemap": page.query_selector(".treemap-container") is not None,
            "call_list": len(page.query_selector_all(".call-item")) > 0,
            "legend": page.query_selector(".legend") is not None,
        }

        print("\n--- Verification ---")
        for name, passed in checks.items():
            status = "PASS" if passed else "FAIL"
            print(f"  [{status}] {name}")

        browser.close()
        print("\nAll screenshots saved to screenshots/")


if __name__ == "__main__":
    main()

File: silent-failure-detector/src/test-dashboard.js
const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');

const SCREENSHOTS_DIR = path.join(__dirname, '..', 'screenshots');
fs.mkdirSync(SCREENSHOTS_DIR, { recursive: true });

async function testDashboard() {
  const browser = await chromium.launch();
  const page = await browser.newPage({ viewport: { width: 1280, height: 800 } });

  console.log('[Test] Opening dashboard...');
  await page.goto('http://localhost:3456/dashboard');
  await page.waitForTimeout(500);

  // Screenshot 1: Empty state
  await page.screenshot({ path: path.join(SCREENSHOTS_DIR, '01-empty-state.png'), fullPage: true });
  console.log('[Test] Screenshot: 01-empty-state.png');

  // Click "Run Demo" button
  console.log('[Test] Clicking "Run Demo"...');
  await page.click('text=Run Demo');
  await page.waitForTimeout(1000);

  // Screenshot 2: After demo run
  await page.screenshot({ path: path.join(SCREENSHOTS_DIR, '02-failures-detected.png'), fullPage: true });
  console.log('[Test] Screenshot: 02-failures-detected.png');

  // Verify failure count is displayed
  const totalBadge = await page.textContent('#totalBadge');
  console.log(`[Test] Total failures shown: ${totalBadge}`);
  if (parseInt(totalBadge) >= 3) {
    console.log('[Test] PASS: Failures detected and shown');
  } else {
    console.log('[Test] FAIL: Expected >= 3 failures');
  }

  // Click "Endpoint Stats" tab
  await page.click('text=Endpoint Stats');
  await page.waitForTimeout(300);
  await page.screenshot({ path: path.join(SCREENSHOTS_DIR, '03-endpoint-stats.png'), fullPage: true });
  console.log('[Test] Screenshot: 03-endpoint-stats.png');

  // Click "Rules" tab
  await page.click('text=Rules');
  await page.waitForTimeout(300);
  await page.screenshot({ path: path.join(SCREENSHOTS_DIR, '04-rules.png'), fullPage: true });
  console.log('[Test] Screenshot: 04-rules.png');

  // Verify rules are shown
  const rulesText = await page.textContent('#rulesTable');
  if (rulesText.includes('order-db-write') && rulesText.includes('payment-processing')) {
    console.log('[Test] PASS: Rules displayed correctly');
  } else {
    console.log('[Test] FAIL: Rules not displayed');
  }

  await browser.close();
  console.log('[Test] Dashboard test complete');
}

testDashboard().catch(err => {
  console.error('[Test] Failed:', err.message);
  process.exit(1);
});
